src/components/album/AlbumsList.js

```diff
@@ -22,41 +24,47 @@ import {
 
 const AlbumsList = () => {
   const [albums, setAlbums] = useState([]); // for albums
-  const [limit, setLimit] = useState(1); // for pagination
+  const [limit, setLimit] = useState(12); // for pagination
   const [searchType, setSearchType] = useState('albumName'); //default value for select input
   const [searchTerm, setSearchTerm] = useState(''); // for search input filed value
   const [message, setMessage] = useState(''); // for not found message
   const [errorMessage, setErrorMessage] = useState(''); // for error message
   const [wishListId, setWishListId] = useState();
+  const [currentPage, setCurrentPage] = useState(1); // Track current page
+  const [lastSearchTerm, setLastSearchTerm] = useState('');
 
   //make an API call with search values to backend and return the result
-  const fetchAlbums = async (searchType, searchTerm, limit) => {
-    try {
-      const response = await axios.get(
-        `${process.env.REACT_APP_API_BASE_PATH}/albums/filter`,
-        {
-          params: {
-            limit,
-            [searchType]: searchTerm,
-          },
+  const fetchAlbums = useCallback(
+    async (searchType, searchTerm, page = 1) => {
+      try {
+        const offset = (page - 1) * limit;
+        const response = await axios.get(
+          `${process.env.REACT_APP_API_BASE_PATH}/albums/filter`,
+          {
+            params: {
+              limit: limit,
+              [searchType]: searchTerm,
+              offset,
+            },
+          }
+        );
+        const searchResult = response.data.albums;
+        setAlbums(searchResult);
+        if (searchResult.length === 0) {
+          setMessage('No result found');
         }
-      );
-      const searchResult = response.data.albums;
-      setAlbums(searchResult);
-      if (searchResult.length === 0) {
-        setMessage('No result found');
+      } catch (error) {
+        console.error('Error fetching albums:', error);
+        setErrorMessage(error.message);
+        setOpen(true);
       }
-    } catch (error) {
-      console.error('Error fetching albums:', error);
-      setErrorMessage(error.message);
-      setOpen(true);
-    }
-  };
+    },
+    [/*currentPage,*/ limit]
+  );
 
-  //display 10 albums when first user visit this page
   useEffect(() => {
-    fetchAlbums('albumName', '', 10);
-  }, []);
+    fetchAlbums(searchType, searchTerm); // Use current search term
+  }, [fetchAlbums, searchType, searchTerm]); // Refetch albums when the page changes
```

Comment:
This will run the `fetchAlbum` call, any time the user types into the search box, even before they click the "Search" button. This is fine, but if that's the case then it seems like having the search button is a bit redundant. Either we remove the search button, or we update this useEffect to use `lastSearchTerm` instead:

```suggestion
    fetchAlbums(searchType, lastSearchTerm); // Use current search term
  }, [fetchAlbums, searchType, lastSearchTerm]); // Refetch albums when the submitted search changes
```

If we do that, we should make sure to update `handleClear` to also setLastSearchTerm to an empty string, so that the fetchAlbum call is triggered in that case as well

Reply:
Code fixed according to them recommendations